### ISSUE TITLE
Questions for align_68.py

### ISSUE THREAD

**Issue by Yusics** [Hangz-nju-cuhk/Talking-Face_PC-AVS#10]
I really appreciate the authors for the great work! The demo looks super well and the models are robustness.

I have two questions for some details in the align_68.py. 
(1) I'm just curious what's the purpose of adding the bias [here](https://github.com/Hangz-nju-cuhk/Talking-Face_PC-AVS/blob/main/scripts/align_68.py#L83) 
(2) Why do you calculate the average for the three points [here](https://github.com/Hangz-nju-cuhk/Talking-Face_PC-AVS/blob/main/scripts/align_68.py#L70). Wouldn't it be better to store the three points for each frame?

Thanks for the great work again!

**Comment by Hangz-nju-cuhk** [comment]
Hi, thank you for looking into the details of the code! Most previous work for facial image generation normally align each face according to the three points (As I have done in my previous paper [https://github.com/Hangz-nju-cuhk/Talking-Face-Generation-DAVS](url)). However, this will lead to a zoom-out-and-in artifact due to the affine transformation. 

Thus in this work, I do not align the samples provided in VoxCeleb2 and choose to align the faces according to the average of all key points in other videos. The bias is to ensure that the face is almost at the center of the cropped frame but  certain misalignment is allowed.

**Comment by Yusics** [comment]
Thank you for your reply! Your reply answered my question! 

Sorry for another question about the training details. Since no preprocessed code is released. I have a question regarding it.
During the training stage, I'm wondering did you train the entire clip or did you sample a few number of frames in a clip and each time you random sample a pair of frames as the training data? Thanks.

**Comment by Hangz-nju-cuhk** [comment]
Hi, for each epoch we sample like 12 continuous frames for contrastive learning and among them, 4 are used for reconstruction training.

**Comment by Yusics** [comment]
Thanks for your prompt reply, it's really helpful. 
Sorry for bring up another question. Is it possible to share the training hyperparameters of all the models? (Batch size, learning rate,.. etc). The hyperparameters would be super helpful if I want to train the model based on the repo. Thanks!
